g11n-ws/modules/md-service-authen/src/main/java/com/vmware/vip/core/auth/interceptor/AuthInterceptor.java

```diff
@@ -37,38 +32,36 @@ public AuthInterceptor(String allowSourceCollection, TokenService tokenService)
 	@Override
 	public boolean preHandle(final HttpServletRequest request, final HttpServletResponse response, final Object handler)
 			throws Exception {
+		final String token = request.getHeader(ConstantsKeys.CSP_AUTH_TOKEN);
 		if (request.getMethod().equalsIgnoreCase(HttpMethod.PUT.name())){
-			UpdateTranslationDTO updateTranslationDTO = objectMapper.readValue(request.getInputStream(), UpdateTranslationDTO.class);
-            request.setAttribute(ConstantsKeys.UPDATEDTO, updateTranslationDTO);
-		    if (updateTranslationDTO.getRequester().equals(ConstantsKeys.VL10N)){
+		    if (!StringUtils.isEmpty(token) && token.equals(ConstantsKeys.VL10N)){
                 return true;
 			}else{
-				return validateCspToken(request, response);
+				return validateCspToken(token, response);
 			}
-
 		}
+
 		if (StringUtils.equalsIgnoreCase(request.getParameter(ConstantsKeys.COLLECT_SOURCE), ConstantsKeys.TRUE)) {
-			PrintWriter writer = response.getWriter();
 			if (allowSourceCollection.equalsIgnoreCase(ConstantsKeys.TRUE)) {
-				return validateCspToken(request,response);
+				return validateCspToken(token,response);
 			} else {
 				response.setStatus(HttpStatus.FORBIDDEN.value());
-				writer.write(this.buildRespBody(HttpStatus.FORBIDDEN.value(), ConstantsKeys.SOURCE_COLLECTION_ERROR));
+				response.getWriter().write(this.buildRespBody(HttpStatus.FORBIDDEN.value(), ConstantsKeys.SOURCE_COLLECTION_ERROR));
 				return false;
 			}
 		}
 		return true;
 	}
 
-	private boolean validateCspToken(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
-		final String token = request.getHeader(ConstantsKeys.CSP_AUTH_TOKEN);
+	private boolean validateCspToken(final String token, final HttpServletResponse response) throws IOException {
 		if(token == null) {
+			response.setContentType("application/json;charset=utf-8");
 			response.setStatus(HttpStatus.UNAUTHORIZED.value());
 			response.getWriter().write(this.buildRespBody(HttpStatus.UNAUTHORIZED.value(), ConstantsKeys.TOKEN_VALIDATION_ERROR));
 			return false;
 		}
 		if (!tokenService.isTokenValid(token)) {
-			// The user is not authenticated.
+			response.setContentType("application/json;charset=utf-8");
```

Comment:
use the string in ConstantsKeys.CONTENT_TYPE_JSON?

Reply:
done